diff --git a/node_modules/calendar-utils/calendar-utils.d.ts b/node_modules/calendar-utils/calendar-utils.d.ts
index 13e9c53..a628e9b 100644
--- a/node_modules/calendar-utils/calendar-utils.d.ts
+++ b/node_modules/calendar-utils/calendar-utils.d.ts
@@ -34,9 +34,11 @@ export interface EventAction {
 }
 export interface CalendarEvent<MetaType = any> {
     id?: string | number;
+    userId? : number;
     start: Date;
     end?: Date;
     title: string;
+    ou? : string;
     color?: EventColor;
     actions?: EventAction[];
     allDay?: boolean;
@@ -133,9 +135,12 @@ export interface GetWeekViewArgs {
     dayEnd: Time;
     weekendDays?: number[];
     segmentHeight: number;
+    segmentWidth?: number;
     viewStart?: Date;
     viewEnd?: Date;
     minimumEventHeight?: number;
+    minimumEventWidth?: number;
+    is_horizontal_dayview?: boolean;
 }
 export declare function getDifferenceInDaysWithExclusions(dateAdapter: DateAdapter, { date1, date2, excluded }: {
     date1: Date;
